File: java/src/menu/components/PasswordField.java
package menu.components;

import java.awt.Color;

import core.Screen;
import menu.ComponentContainer;
import misc.math.pixel.PixelRect;

public class PasswordField extends EditField
{
	public PasswordField(ComponentContainer parent, PixelRect rect)
	{
		super(parent, rect);
	}

	@Override public void render()
	{
		Screen.g().setColor(Color.BLUE);
		Screen.g().fillRect(getOffset().getX(), getOffset().getY(), getWidth(), getHeight());
		Screen.g().setColor(Color.BLACK);
		Screen.g().drawString(getDotString(), getOffset().getX() + 1, getOffset().getY() + 12);
	}

	private String getDotString()
	{
		String dotString = "";
		for (int i = 0; i < getText().length(); i++)
		{
			dotString += "*";
		}
		return dotString;
	}
}

File: cpp/src/menu/components/PasswordField.cpp
#include "PasswordField.hpp"

#include "../../misc/Debug.hpp"
#include "../../core/Screen.hpp"

PasswordField::PasswordField(ComponentContainer* parent, const PixelRect& rect, const std::string& s) : MenuComponent(parent, rect), text(s)
{

}

PasswordField::PasswordField(ComponentContainer* parent, const PixelRect& rect) : MenuComponent(parent, rect), text("")
{

}

void PasswordField::onTextEntered(char c)
{
	if (c == 8)
	{
		text = text.substr(0, text.length()-1);
	}
	else
	{
		text += c;
	}
}

void PasswordField::render() const
{
	Screen::fillRect(getAbsoluteRect(), sf::Color::Blue);
	std::string renderedText = "";
	for (unsigned int i = 0; i < getText().length(); i++)
	{
		renderedText += "*";
	}

	Screen::drawText(renderedText, getAbsoluteRect().getPosition(), sf::Color::Black);
}

const std::string& PasswordField::getText() const { return text; }
